### DISCUSSION TITLE
Create one point service for all sub component.

### DISCUSSION THREAD

**Discussion by jitendraP-ashutec** [lit/lit#3326] [Q&A] [answered]
I want to create [smart-table](https://akveo.github.io/ng2-smart-table/#/) like functionality. I want to create one service which can share original data to all sub component used by web component. like search, filter, sort all can use same data(same service as we create singleton service in Angular)

What is best way for this.

Thanks,
Jitendra Prajapati

**Comment by usergenic** [comment, marked as answer]
We have been developing a package that is similar in some ways to the notion of singleton service and dependency injection in Angular, that may address your needs.  Check out the README for our `@lit-labs/context` package here: https://github.com/lit/lit/tree/main/packages/labs/context

**Reply by jitendraP-ashutec** [reply]
Hello @usergenic 

can I use `contextProvider` in inner component to set value, or I have to set value from it root component ?  If not then have any other way to pass to child to parent ?

` @contextProvider({ context: loggerContext })
  loggerContext: any =  { a: 'a' }; `
